### PR TITLE
[log-shipper] Fixed reloader script

### PR DESCRIPTION
**Pull request by zuzzas**
## Description

Fixed Go reloader implementation.

## Why do we need it, and what problem does it solve?

The previous version did not work.

## What is the expected result?

The reloader script works properly and displays a diff.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: log-shipper
type: fix
summary: Fixed vector-reloader
impact_level: low
```

### CODE REVIEW

File: modules/460-log-shipper/images/vector/main.go [hunk 1]
```diff
@@ -92,11 +90,17 @@ func main() {
 	<-make(chan struct{})
 }
 
-func reloadVectorConfig() error {
+func reloadVectorConfig() (err error) {
 	tempConfigDir, err := os.MkdirTemp("", "vector-config")
```
Comment (nabokihms) [inline modules/460-log-shipper/images/vector/main.go:94]: It should be fixed, I pointed to the problem when we were discussing it internally. 
Files are created in the tmp dir and deleted only after the pod restart.

**Review by nabokihms — CHANGES_REQUESTED** [review]

File: modules/460-log-shipper/images/vector/main.go [hunk 2]
```diff
@@ -155,7 +159,54 @@ func shouldReload(tempConfigDir string) (bool, error) {
 		return false, err
 	}
 
-	return oldChecksum != newChecksum, nil
+	log.Print("old cksum ", oldChecksum)
+	log.Print("new cksum ", newChecksum)
+
+	if oldChecksum != newChecksum {
```
[the remark this answers is not in the document]
Comment (zuzzas) [reply modules/460-log-shipper/images/vector/main.go]: That was before un-drafting the PR.

File: modules/460-log-shipper/images/vector/main.go [hunk 3]
```diff
@@ -187,5 +239,7 @@ func getFileChecksum(path string) (string, error) {
 		return "", err
 	}
 
+	log.Printf("cksum for %s: %s", path, hash.Sum(nil))
```
[the remark this answers is not in the document]
Comment (zuzzas) [reply modules/460-log-shipper/images/vector/main.go]: That was before un-drafting the PR.

File: modules/460-log-shipper/images/vector/main.go (hunk 1 above)
Comment (zuzzas) [reply modules/460-log-shipper/images/vector/main.go:95]: It is fixed by introducing a defer with cleanup.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Pull request merged**